cloud/scope/powervs_cluster_test.go

```diff
@@ -1570,3 +1573,415 @@ func getPowerVSClusterScopeParams() PowerVSClusterScopeParams {
+		clusterScopeParams.PowerVSClientFactory = func() (powervs.PowerVS, error) {
+			mockPowerVS.EXPECT().GetNetworkByName(*network.Name).Return(network, nil)
+			return mockPowerVS, nil
+		}
+
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+
+		networkID, err := clusterScope.getNetwork()
+		g.Expect(err).To(BeNil())
+		g.Expect(networkID).To(Equal(network.NetworkID))
+	})
+
+	t.Run("When network ID and DHCPServer is empty and GetNetworkByName returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.IBMPowerVSCluster.Spec.Network = infrav1beta2.IBMPowerVSResourceReference{}
+		clusterScopeParams.IBMPowerVSCluster.Name = clusterName
+		networkName := fmt.Sprintf("DHCPSERVER%s_Private", clusterScopeParams.IBMPowerVSCluster.Name)
+
+		clusterScopeParams.PowerVSClientFactory = func() (powervs.PowerVS, error) {
+			mockPowerVS.EXPECT().GetNetworkByName(networkName).Return(nil, fmt.Errorf("GetNetworkByName error"))
+			return mockPowerVS, nil
+		}
+
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+
+		networkID, err := clusterScope.getNetwork()
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(networkID).To(BeNil())
+	})
+
+	t.Run("When network ID and DHCPServer is empty and GetNetworkByName returns empty network", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.IBMPowerVSCluster.Spec.Network = infrav1beta2.IBMPowerVSResourceReference{}
+		clusterScopeParams.IBMPowerVSCluster.Name = clusterName
+		networkName := fmt.Sprintf("DHCPSERVER%s_Private", clusterScopeParams.IBMPowerVSCluster.Name)
+		clusterScopeParams.PowerVSClientFactory = func() (powervs.PowerVS, error) {
+			mockPowerVS.EXPECT().GetNetworkByName(networkName).Return(nil, nil)
+			return mockPowerVS, nil
+		}
+
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+
+		networkID, err := clusterScope.getNetwork()
+		g.Expect(err).To(BeNil())
+		g.Expect(networkID).To(BeNil())
+	})
+}
+
+func TestIsDHCPServerActive(t *testing.T) {
+	var (
+		mockPowerVS *mockP.MockPowerVS
+		mockCtrl    *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockPowerVS = mockP.NewMockPowerVS(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+
+	t.Run("When DHCPID is empty", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.IBMPowerVSCluster.Spec.DHCPServer.ID = ptr.To("")
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+
+		isActive, err := clusterScope.isDHCPServerActive()
+		g.Expect(err).To(Equal(fmt.Errorf("DHCP ID is empty")))
+		g.Expect(isActive).To(BeFalse())
+	})
+	t.Run("When GetDHCPServer returns error", func(t *testing.T) {
```

Comment:
I think this should be
```suggestion
	t.Run("When checkDHCPServerStatus returns error", func(t *testing.T) {
```
right?